Lake/Build/Module.lean

```diff
@@ -61,7 +61,7 @@ def recBuildPrecompileDynlibs (pkg : Package) (imports : Array Module)
   let mut jobs := #[]
   for imp in imports do
     if imp.shouldPrecompile then
-      let (_, transImports) ← imp.imports.fetch
+      let transImports ← imp.transImports
```

Comment:
If precompilation is on and A imports B and C who both import D who, in turn, imports some arbitrary module tree, then this change will compute the transitive imports for D in both B and C. It will then be recompute them again when building the dynlib for the import in `recBuildDynlib` below. Given your observation of the performance costs of this computation, I would suggest caching `transImports` in the build store through a new facet (like the current `imports` one).